adobe/spectrum-css
# feat(coachindicator, coachmark): add new component & migrate to spectrum-tokens

## Description

Recreation of #1992, view previous PR for full code review, regression testing, and comments. 

Migrates tokens and implements new styling for coach mark

Splits the coach mark indicator and coach mark popover into 2 separate components, CoachIndicator and CoachMark
is "responsive" and displays different button and menu styling for Large scale (mobile)
Updates stories for both components



## How and where has this been tested?

Please tag yourself on the tests you've marked complete to confirm the tests have been run by someone other than the author.

### Validation steps
#### Test Outline @mdt2
1. Open the [docs site](https://pr-1992--spectrum-css.netlify.app/coachmark) for the CoachMark component.
- [x] component should display mobile styling when scale is Large
- [x] styling should match XD files 

2. Open the [storybook site](https://pr-1992--spectrum-css.netlify.app/preview/?path=/story/components-coach-mark--default) for the CoachMark component
- [x] controls should all function as expected 
- [x] styling should match docs site 


3. Open the [docs site](https://pr-1992--spectrum-css.netlify.app/coachindicator) for the CoachIndicator component.
- [x] Styling should match the live docs site 

4. Open the [storybook site](https://pr-1992--spectrum-css.netlify.app/preview/?path=/docs/components-coach-indicator--docs) for the CoachIndicator component
- [ ] controls should all function as expected
- [x] styling should match docs site

### Regression testing

Validate:

1. The documentation pages for at least two other components are still loading, including:


2. If components have been modified, VRTs have been run on this branch:


## Screenshots



## To-do list
